[BE] 로그인 시 Refresh Token 중복 이슈

### 🐞버그 설명

- [x] 리프레시 토큰을 DB에 저장 시, ID로 중복된 값이 들어가서 생기는 문제
- [x] 리프레시 토큰 발행 시, EMPTY 가 아닌 UUID로 PAYLOAD를 설정해준다

### 📄 버그 로그

```shell
2023-09-17T16:47:51.665+09:00
Caused by: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry ‘3’ for key ‘refresh_token.UK_dnbbikqdsc2r2cee1afysqfk9’ at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:118) at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122) at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:916) at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1061) at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1009) at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1320) at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:994) at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61) at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java) at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:275) ... 80 common frames omitted
access-error.log
843255971531:2023-map-befine-dev
@ingestionTime
1694936872102
@log
843255971531:2023-map-befine-dev
@logStream
access-error.log
@message
Caused by: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry ‘3’ for key ‘refresh_token.UK_dnbbikqdsc2r2cee1afysqfk9’
at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:118)
at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:916)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1061)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1009)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1320)
at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:994)
at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.executeUpdate(ResultSetReturnImpl.java:275)
... 80 common frames omitted
@timestamp
1694936871665
```


### 🌏 버그 발생 환경

MacOS, Chrome

### ✅ 버그 재현 방법

_No response_

### 📎 참고 자료

_No response_

### ⏰ 추정 시간

비관적 : 2시간
낙관적 : 1시간

--- a/backend/src/main/java/com/mapbefine/mapbefine/auth/infrastructure/JwtTokenProvider.java
+++ b/backend/src/main/java/com/mapbefine/mapbefine/auth/infrastructure/JwtTokenProvider.java
@@ -11,13 +11,12 @@ import io.jsonwebtoken.JwtException;
 import io.jsonwebtoken.Jwts;
 import io.jsonwebtoken.SignatureAlgorithm;
 import java.util.Date;
+import java.util.UUID;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Component;
 
 @Component
 public class JwtTokenProvider implements TokenProvider {
-
-    private static final String EMPTY = "";
 
     private final String secretKey;
     private final long accessExpirationTime;
@@ -41,7 +40,9 @@ public class JwtTokenProvider implements TokenProvider {
     }
 
     public String createRefreshToken() {
-        return createToken(EMPTY, refreshExpirationTime);
+        UUID payload = UUID.randomUUID();
+
+        return createToken(payload.toString(), refreshExpirationTime);
     }
 
     private String createToken(String payload, Long validityInMilliseconds) {

--- a/backend/src/main/java/com/mapbefine/mapbefine/common/config/WebConfig.java
+++ b/backend/src/main/java/com/mapbefine/mapbefine/common/config/WebConfig.java
@@ -16,6 +16,7 @@ public class WebConfig implements WebMvcConfigurer {
     public void addCorsMappings(CorsRegistry registry) {
         registry.addMapping("/**")
                 .allowedOrigins("http://localhost:3000", "https://mapbefine.kro.kr", "https://mapbefine.com")
+                .allowedHeaders("refresh-token")
                 .allowedMethods("*")
                 .exposedHeaders(LOCATION, SET_COOKIE);
     }

--- a/backend/src/main/java/com/mapbefine/mapbefine/common/config/WebConfig.java
+++ b/backend/src/main/java/com/mapbefine/mapbefine/common/config/WebConfig.java
@@ -16,9 +16,7 @@ public class WebConfig implements WebMvcConfigurer {
     public void addCorsMappings(CorsRegistry registry) {
         registry.addMapping("/**")
                 .allowedOrigins("http://localhost:3000", "https://mapbefine.kro.kr", "https://mapbefine.com")
-                .allowedHeaders("refresh-token")
                 .allowedMethods("*")
-                .allowCredentials(true)
                 .exposedHeaders(LOCATION, SET_COOKIE);
     }
 

